### PR TITLE
feat: support v-on="obj"

### PR DESCRIPTION
**Pull request by Jevon617**

### CODE REVIEW

File: packages/compiler-vapor/src/ir.ts [hunk 1]
```diff
@@ -94,6 +95,12 @@ export interface SetDynamicPropsIRNode extends BaseIRNode {
   props: IRProps[]
 }
 
+export interface SetDynamicEventsIRNode extends BaseIRNode {
+  type: IRNodeTypes.SET_DYNAMIC_EVENTS
+  element: number
+  prop: SimpleExpressionNode
```
Comment (sxzz) [inline packages/compiler-vapor/src/ir.ts:101]:
```suggestion
  event: SimpleExpressionNode
```

**Review by sxzz — CHANGES_REQUESTED** [review]
We should avoid introducing the `setDynamicEvents` function and instead reuse `setDynamicProps` to support the following code:

```vue
<button v-bind="{ onClick: fn }" />
<button v-on="{ click: fn }" />
```

Both approaches are equivalent.

**Review by sxzz — CHANGES_REQUESTED** [review]
Should also emit an error, if no `modifiers` or no `arg`

```vue
<button v-on.prevent />
```

**Comment by Jevon617** [comment]
> Should also emit an error, if no `modifiers` or no `arg`
> 
> ```
> <button v-on.prevent />
> ```

Are you saying that this situation should emit an error?
```
<button v-on.prevent="{}" />
```

**Comment by Jevon617** [comment]
I will replace `setDynamicEvents` with `setDynamicProps` through the next pr, and add some test cases.

**Pull request merged**